flow/project.py

```diff
@@ -418,11 +435,21 @@ def __init__(self, cmd, pre=None, post=None, directives=None):
     def __str__(self):
         return "{type}(cmd='{cmd}')".format(type=type(self).__name__, cmd=self._cmd)
 
-    def eligible(self, job):
-        "Eligible, when all pre-conditions are true and at least one post-condition is false."
-        pre = all(cond(job) for cond in self._prereqs)
+    def eligible(self, job, ignore_conditions='none'):
+        """Eligible, when all pre-conditions are true and at least one post-condition is false,
+        or corresponding conditions are ignored
+        """
+        if ignore_conditions in _IGNORE_CONDITIONS.keys():
```

Comment:
Once you change the API, the correct test would be `if type(ignore_conditions) == IgnoreConditions`. @b-butler do you think this test is necessary, or would you prefer to allow users to pass in integers directly and ducktype them? My initial thought was that ducktyping is fine, but in Python maybe it's asking too much of users to not misuse enums, which are far less common in Python than in lower-level languages.